src/datepicker/ngb-date.ts

```diff
@@ -0,0 +1,25 @@
+export class NgbDate {
+  static from(date: {year: number, month: number, date?: number}) {
+    return date ? new NgbDate(date.year, date.month, date.date ? date.date : 1) : null;
+  }
+
+  constructor(public year: number, public month: number, public date: number) {}
+
+  equals(other: NgbDate) {
+    return other && this.year === other.year && this.month === other.month && this.date === other.date;
+  }
+
+  before(other: NgbDate) {
+    return !other ? false : this.year === other.year ?
+                    this.month === other.month ? this.date === other.date ? false : this.date < other.date :
+                                                 this.month < other.month :
+                    this.year < other.year;
+  }
+
+  after(other: NgbDate) {
+    return !other ? false : this.year === other.year ?
+                    this.month === other.month ? this.date === other.date ? false : this.date > other.date :
+                                                 this.month > other.month :
+                    this.year > other.year;
+  }
+}
```

Comment:
Those before / after methods are kind of long-winded. Maybe it would be easier to rewrite them with if / else instead of using tenary?

Other thought: how about implementing a `compare`   method with standard contract (-1, 0, 1 as return values). Not saying we need to do this in the PR, just sth to think about.